class.jetpack.php

```diff
@@ -5344,6 +5347,36 @@ public static function validate_sync_error_idc_option() {
 	}
 
 	/**
+	 * Gets the value that is to be saved in the jetpack_sync_error_idc option.
+	 *
+	 * @since 4.4.0
+	 *
+	 * @return array {
+	 *     @type string 'home'    The current home URL.
+	 *     @type string 'siteurl' The current site URL.
+	 * }
+	 */
+	public static function get_sync_error_idc_option() {
+		$options = array(
+			'home'    => get_option( 'home' ),
+			'siteurl' => get_option( 'siteurl' ),
+		);
+
+		$returned_values = array();
+		foreach( $options as $key => $option ) {
+			$parsed_url = parse_url( trailingslashit( $option ) );
```

Comment:
`parse_url()` doesn't actually give a `host` value if an IP address (or anything without a protocol) is passed, which would throw a notice below when you use it.  It needs a protocol, so maybe you could (pre)normalize the URL by running it through `esc_url_raw()` or something, and then pass it through `wp_parse_url()`.